src/middleware/errorHandler.js

```diff
@@ -151,67 +175,71 @@ const filterSecrets = (error, req, res, next) => {
 		req.body = filter(req.body);
 		error.data = filter(error.data);
 		error.options = filter(error.options);
+		error.params = filter(error.params);
 	}
 	next(error);
 };
 
-const saveResponseFilter = (error) => ({
-	name: error.name,
-	message: error.message instanceof Error && error.message.message ? error.message.message : error.message,
-	code: error.code,
-	traceId: error.traceId,
-});
+const createErrorDetailTO = (status, type, title, detail, customFields) => {
+	return { status, type, title, detail, ...customFields };
+};
 
-const sendError = (res, error) => {
-	res.status(error.code).json(saveResponseFilter(error));
+const getErrorResponseFromBusinessError = (businessError) => {
+	const customFields = {};
+	let status = 500;
+	const { message: type, title, defaultMessage: detail } = businessError;
+	if (businessError instanceof ValidationError || businessError instanceof AssertionError) {
+		status = 400;
+		Object.assign(customFields, { validation_errors: businessError.params });
+	} else if (businessError instanceof DocumentNotFound) {
+		status = 404;
+	}
+	return createErrorDetailTO(status, type, title, detail, customFields);
 };
 
-const handleSilentError = (error, req, res, next) => {
-	if (error instanceof SilentError || (error && error.error instanceof SilentError)) {
-		if (Configuration.get('SILENT_ERROR_ENABLED')) {
-			res.append('x-silent-error', true);
+const getErrorResponse = (error, req, res, next) => {
+	let errorDetail;
+	if (isSilentError(error)) {
+		if (Configuration.get('SILENT_ERROR_ENABLED') === true) {
+			res.append('x-silent-error', true); // TODO is removed in production?
 		}
-		res.status(200).json({ success: 'success' });
+		// do not return this as error via REST
+		return res.status(200).json(SilentError.RESPONSE_CONTENT);
+	}
+	if (isApplicationError(error)) {
+		// Application Errors
+		errorDetail = getErrorResponseFromBusinessError(error);
+	} else if (isFeatherError(error)) {
+		// Framework Errors
+		const { code, className: type, name: title, message: detail } = error;
+		errorDetail = createErrorDetailTO(code, type, title, detail);
 	} else {
-		next(error);
+		// Unhandled Errors
+		const unhandledError = new InternalServerError(error);
+		const { message: type, title, defaultMessage: detail } = unhandledError;
+		errorDetail = createErrorDetailTO(500, type, title, detail);
 	}
+
+	res.status(errorDetail.status).json(errorDetail);
 };
 
-const handleValidationError = (error, req, res, next) => {
+const convertOpenApiValidationError = (error) => {
 	// todo: handle other validation errors so they are formatted properly
 	let err = error;
 	if (error instanceof OpenApiValidator.error.NotFound) {
+		logger.debug('Open API Validation path is missing!');
 		err = new PageNotFound(error);
 	} else if (err instanceof OpenApiValidator.error.BadRequest) {
-		err = new BadRequest(error);
+		err = new ValidationError(API_VALIDATION_ERROR, err.errors);
```

Comment:
is there a conversion of err.errors needed?

Reply:
nope - the err.errors provides OpenAPIValidation specific errors